apis/placement/v1beta1/stageupdate_types.go

```diff
@@ -186,8 +186,8 @@ type UpdateRunSpec struct {
 
 	// The resource snapshot index of the selected resources to be updated across clusters.
 	// The index represents a group of resource snapshots that includes all the resources a ResourcePlacement selected.
-	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="resourceSnapshotIndex is immutable"
+	// +kubebuilder:validation:Optional
```

Comment:
Did you test this in an e2e env? Please paste your test updateRun obj (including both spec and status) in PR description.